gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GCSVeneerImplTest.java

```diff
@@ -0,0 +1,156 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.cloud.hadoop.gcsio.testing.MockGoogleCloudStorageImplFactory.mockedGcs;
+import static com.google.cloud.hadoop.gcsio.testing.MockGoogleCloudStorageImplFactory.mockedGcsVeneer;
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.arbitraryInputStreamSupplier;
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.inputStreamResponse;
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.jsonErrorResponse;
+import static com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.mockTransport;
+import static com.google.common.net.HttpHeaders.CONTENT_LENGTH;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.api.services.storage.Storage;
+import com.google.auth.Credentials;
+import com.google.cloud.hadoop.util.HttpTransportFactory;
+import com.google.cloud.hadoop.util.RetryHttpInitializer;
+import com.google.cloud.hadoop.util.RetryHttpInitializerOptions;
+import com.google.cloud.hadoop.util.testing.MockHttpTransportHelper.ErrorResponses;
+import com.google.cloud.hadoop.util.testing.ThrowingInputStream;
+import java.io.IOException;
+import java.nio.channels.ClosedByInterruptException;
+import java.nio.channels.WritableByteChannel;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests that require a particular configuration of GoogleCloudStorageImpl. */
+@RunWith(JUnit4.class)
+public class GCSVeneerImplTest {
+
+  private static final String BUCKET_NAME = "foo-bucket";
+  private static final String OBJECT_NAME = "bar-object";
+
+  /**
+   * Test handling of various types of Errors thrown during JSON API call for
+   * GoogleCloudStorage.create(2).
+   */
+  @Test
+  public void testCreateObjectApiError() throws IOException {
+    // Set up the mock Insert to throw an exception when execute() is called.
+    Error fakeError = new Error("Fake error");
+    MockHttpTransport transport =
+        mockTransport(
+            jsonErrorResponse(ErrorResponses.NOT_FOUND),
+            inputStreamResponse(
+                CONTENT_LENGTH,
+                /* headerValue = */ 1,
+                new ThrowingInputStream(/* readException = */ null, fakeError)));
+    GoogleCloudStorageImpl gcs = mockedGcs(transport);
```

Comment:
why is this not Veneer?

Reply:
Yeah, found many discrepancies here will fix it and also try to increase the coverage.